test/helpers/value_hook.dart

```diff
@@ -1,42 +1,198 @@
 import 'dart:async';
 
-import 'package:test/test.dart' as test show expect, Matcher;
-
-import 'wait_for.dart';
+import 'package:test/test.dart' as test show expect, Matcher, TestFailure;
 
 /// Useful to easily test handler functions
 /// that set values outside of their scope.
 ///
 /// ```dart
-/// // Construct VHook with initial value of 'false'
-/// VHook<bool> calledHandler = VHook<bool>(false);
+/// // Construct VHook with no initial value
+/// VHook calledHandler = VHook();
 ///
 /// willCallHandler(() {
-///   calledHandler.set(true);
 ///   // Some handler code
+///   // ...
+///   calledHandler.complete();
 /// });
 ///
 /// // Test that the handler was called
-/// // This will throw an error if the handler was not called.
-/// calledHandler.expect(equals(true));
+/// // If the handler was not called after 5s, throw an error
+/// calledHandler.awaitCompletion(Duration(seconds: 5));
 /// ```
 class VHook<T> {
   /// The value of the variable.
-  T _value;
+  dynamic _value;
+
+  /// The error that might be completed with
+  Object _error = _VHookValue.none;
```

Comment:
Why not using an optional here? Would make no sense to have an error object of type `Null`. Is `null` even derived from `Object`?

Reply:
I dont thinck that it is derived from Object, i'll look into changing this to `Object?`